pkg/egressservice/egressservice_test.go

```diff
@@ -0,0 +1,515 @@
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		var svcName string
+
+		if testCase.expectedError != nil {
+			svcName = "hello"
+		} else {
+			svcName = testCase.name
+		}
+
+		testEgressService := buildDummyEgressService(svcName, testCase.nsname, testCase.sourceIPBy)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testEgressService)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects:  runtimeObjects,
+				SchemeAttachers: testSchemes,
+			})
+		}
+
+		testEgressServiceBuilder := buildDummyEgressServiceBuilder(
+			testSettings, testCase.name, testCase.nsname, testCase.sourceIPBy)
+
+		testEgressService, err := testEgressServiceBuilder.Get()
+
+		if testCase.expectedError != nil {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+			assert.Empty(t, testEgressService)
+		} else {
+			assert.Equal(t, testCase.name, testEgressService.ObjectMeta.Name)
+			assert.Equal(t, testCase.nsname, testEgressService.ObjectMeta.Namespace)
+			assert.Equal(t, testCase.sourceIPBy, string(testEgressService.Spec.SourceIPBy))
+		}
+	}
+}
+
+func TestUpdate(t *testing.T) {
+	testCases := []struct {
+		name          string
+		nsname        string
+		sourceIPBy    string
+		newVrfNetwork string
+		nodeSelector  map[string]string
+		expectedError error
+	}{
+		{
+			name:          egressTestSvcName,
+			nsname:        egressTestSvcNamespace,
+			sourceIPBy:    "LoadBalancerIP",
+			newVrfNetwork: "5678",
+			nodeSelector:  map[string]string{},
+			expectedError: nil,
+		},
+		{
+			name:          egressTestSvcName,
+			nsname:        egressTestSvcNamespace,
+			sourceIPBy:    "LoadBalancerIP",
+			newVrfNetwork: "",
+			nodeSelector:  map[string]string{"role": "infrastructure"},
+			expectedError: nil,
+		},
+		{
+			name:          egressTestSvcName,
+			nsname:        egressTestSvcNamespace,
+			sourceIPBy:    "LoadBalancerIP",
+			newVrfNetwork: "2468",
+			nodeSelector: map[string]string{
+				"role": "infrastructure",
+				"zone": "public",
+			},
+			expectedError: nil,
+		},
+		{
+			name:          egressTestSvcName,
+			nsname:        egressTestSvcNamespace,
+			sourceIPBy:    "LoadBalancerIP",
+			newVrfNetwork: " ",
+			nodeSelector:  map[string]string{},
+			expectedError: fmt.Errorf("Cannot use empty VRF network"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		testEgressServiceBuilder, err := Pull(buildTestClientWithDummyEgressService(),
+			testCase.name, testCase.nsname)
+
+		assert.Nil(t, err)
+
+		if testCase.newVrfNetwork != "" {
+			testEgressServiceBuilder = testEgressServiceBuilder.WithVRFNetwork(testCase.newVrfNetwork)
+		}
+
+		if len(testCase.nodeSelector) != 0 {
+			testEgressServiceBuilder = testEgressServiceBuilder.WithNodeLabelSelector(testCase.nodeSelector)
+		}
```

Comment:
Since we're just testing update it can be simpler to just pick one value to update and verify that one value has changed, like [here in the metallb package](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/metallb/adresspool_test.go#L246).

Reply:
updated